java.lang.IndexOutOfBoundsException when clicking on a slot in an inventory of type ANVIL created via Bukkit.createInventory()

Caused by: java.lang.IndexOutOfBoundsException: Index: 7, Size: 0
	at java.util.ArrayList.rangeCheck(ArrayList.java:659) ~[?:1.8.0_412]
	at java.util.ArrayList.get(ArrayList.java:435) ~[?:1.8.0_412]
	at net.minecraft.server.v1_8_R3.Container.getSlot(Container.java:109) ~[patched_1.8.8.jar:git-PandaSpigot-121]
	at net.minecraft.server.v1_8_R3.PlayerConnection.a(PlayerConnection.java:1514) ~[patched_1.8.8.jar:git-PandaSpigot-121]
	at net.minecraft.server.v1_8_R3.PacketPlayInWindowClick.a(PacketPlayInWindowClick.java:17) ~[patched_1.8.8.jar:git-PandaSpigot-121]
	at net.minecraft.server.v1_8_R3.PacketPlayInWindowClick.a(PacketPlayInWindowClick.java:5) ~[patched_1.8.8.jar:git-PandaSpigot-121]
	at net.minecraft.server.v1_8_R3.PlayerConnectionUtils$1.run(PlayerConnectionUtils.java:9) ~[patched_1.8.8.jar:git-PandaSpigot-121]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[?:1.8.0_412]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[?:1.8.0_412]
	at net.minecraft.server.v1_8_R3.SystemUtils.a(SystemUtils.java:11) ~[patched_1.8.8.jar:git-PandaSpigot-121]
	... 6 more

--- a/patches/server/0100-Fix-custom-anvil-inventory.patch
+++ b/patches/server/0100-Fix-custom-anvil-inventory.patch
@@ -1,0 +1,49 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: uRyanxD <familiarodrigues123ro@gmail.com>
+Date: Sun, 7 Jul 2024 20:19:50 -0300
+Subject: [PATCH] Fix custom anvil inventory
+
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/inventory/CraftContainer.java b/src/main/java/org/bukkit/craftbukkit/inventory/CraftContainer.java
+index 1dbc588e72111afebbcfc3b1d714b7f24d6d42fe..97e76be8aafbd68e8d1f055289851bf3828d60b4 100644
+--- a/src/main/java/org/bukkit/craftbukkit/inventory/CraftContainer.java
++++ b/src/main/java/org/bukkit/craftbukkit/inventory/CraftContainer.java
+@@ -139,6 +139,11 @@ public class CraftContainer extends Container {
+         case BREWING:
+             setupBrewing(top, bottom);
+             break;
++        // PandaSpigot start
++        case ANVIL:
++            setupAnvil(top, bottom);
++            break;
++        // PandaSpigot end
+         case HOPPER:
+             setupHopper(top, bottom);
+             break;
+@@ -277,6 +282,26 @@ public class CraftContainer extends Container {
+         // End copy from ContainerBrewingStand
+     }
+ 
++    // PandaSpigot start - This code copied from ContainerAnvil
++    private void setupAnvil(IInventory top, IInventory bottom) {
++        this.a(new Slot(top, 0, 27, 47));
++        this.a(new Slot(top, 1, 76, 47));
++        this.a(new Slot(top, 2, 134, 47));
++
++        int i;
++
++        for (i = 0; i < 3; ++i) {
++            for (int j = 0; j < 9; ++j) {
++                this.a(new Slot(bottom, j + i * 9 + 9, 8 + j * 18, 84 + i * 18));
++            }
++        }
++
++        for (i = 0; i < 9; ++i) {
++            this.a(new Slot(bottom, i, 8 + i * 18, 142));
++        }
++    }
++    // PandaSpigot end
++
+     private void setupHopper(IInventory top, IInventory bottom) {
+         // This code copied from ContainerHopper
+         byte b0 = 51;

